### DISCUSSION TITLE
Quarto Blog listing won't link to individual blog posts once published to web server

### DISCUSSION THREAD

**Discussion by kyledemaria** [quarto-dev/quarto-cli#9657] [Q&A]
### Description

I am trying to publish a Quarto Blog to a private web server. When I do so, the links on the listing page do not direct to the individual blog posts. For example, when I click a blog post on the listings page, I am taken to ".../_site/posts/2024_03/". The result is a 404 page not found error. However, the actual blog post is hosted at ".../_site/posts/2024_03/index.html". For some reason, the listing page is not directing to the index.html file itself but rather pointing to the folder it is within. This issue does not occur on my local machine — only when I try to publish to the web server.

I have recreated the _site folder on the web server, but have only uploaded the listing page and the blog post index.html files. I have also set `self-contained: true` and `embed-resources: true` in the yamls of the listing and blog post files. I'm running Quarto version 1.3.450.

Any help would be greatly appreciated!

Here is the yaml of the listing page.
````qmd
---
title: |
listing:
  contents:
    - posts
  date-format: "MMMM YYYY"
  sort: "date desc"
  type: default
  categories: true
  sort-ui: false
  filter-ui: false
page-layout: full

format:
  html:
    self-contained: true
    embed-resources: true
---
````

Here is the _quarto.yaml file.
````qmd
project:
  type: website

website:
  title: "zzz"
  navbar:
    search: true
    right:
      - text: "zzz"
        href: https://zzz.com
        target: "_blank"
format:
  html:
    title-block-style: default
    theme: flatly
    css: styles.css

editor: visual
````

**Comment by mcanouil** [comment]
`self-contained: true` does the same thing as `embed-resources: true`, but `embed-resources: true` is the correct current option.

Why do you set this option for a listing/website?

**Reply by kyledemaria** [reply]
Thank you so much for the response! I updated the files so as to only include `embed-resources: true` and unfortunately the problem persists. My reasoning to include this option is that my blog pages have .png files in them, and I was hoping to be able to simplify the publishing process by embedding them within the outputted html files. Do you have any other thoughts as to what might be wrong?

**Reply by mcanouil** [reply]
How do you publish and to where?

Assets should be detected by Quarto and published alongside the website, especially for images.

**Reply by kyledemaria** [reply]
So I removed `embed-resources: true` and the links to the posts now work! However, all of the formatting and styling of the pages has been lost, making them unusable in their current state.

One of the other reasons I was trying to use `embed-resources` is that the private web server I'm using does not allow me to upload JavaScript files but we can upload CSS files. I was hoping to get around this by embedding the JavaScript within the HTML files.

Is there a way that those `site_libs` resources (e.g., bootstrap, quarto-html, quarto-listing, etc.) can be pulled from another online source, like a stable location where those JavaScript or CSS files might live?

**Reply by mcanouil** [reply]
The resources are shared across all pages, if you embed everything, it means you duplicate everything.
That's usually a bad idea for a website.

I don't think you can do what you want with the constraints you have, or you'll have to do it the very hard way such as editing all the files to target the `site_libs` elsewhere.
You could try setting `embed-resources` at the project level (`_quarto.yml`).

**Reply by kyledemaria** [reply]
Thanks for this perspective. I tried setting `embed-resources` at the project level and the linking problem returned.

If I were to take the hard route of having the files point to `site_libs` hosted elsewhere (perhaps on GitHub), is there a yaml setting I can add that will tell the documents to look at that location? Or would I actually need to edit the resultant HTML files?

**Reply by mcanouil** [reply]
> is there a yaml setting I can add that will tell the documents to look at that location

That would not be the hard way if there was, isn't it?
